packages/vite/src/node/ssr/__tests__/ssrLoadModule.spec.ts

```diff
@@ -292,3 +292,26 @@ test('plugin error', async () => {
       Plugin: test-plugin"
   `)
 })
+
+test('named exports overwrite export all', async () => {
+  const server = await createDevServer()
+  const mod = await server.ssrLoadModule(
+    './fixtures/named-overwrite-all/main.js',
+  )
+
+  // ESM spec doesn't allow conflicting `export *` and such duplicate exports are removed (in this case "d"),
+  // but this is likely not possible to support due to Vite dev SSR's lazy nature.
+  // [Node]
+  //   $ node -e 'import("./packages/vite/src/node/ssr/__tests__/fixtures/named-overwrite-all/main.js").then(console.log)'
+  //   [Module: null prototype] { a: 'main-a', b: 'dep1-b', c: 'main-c' }
+  // [Rollup]
+  //   Conflicting namespaces: "main.js" re-exports "d" from one of the modules "dep1.js" and "dep2.js" (will be ignored).
+  expect(mod).toMatchInlineSnapshot(`
+    {
+      "a": "main-a",
+      "b": "dep1-b",
+      "c": "main-c",
+      "d": "dep1-d",
+    }
+  `)
```

Comment:
I just learned that ESM spec says duplicate names from `export * from ...` should be removed entirely :open_mouth: (Didn't actually read the spec but it's on MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export#specifications:~:text=If%20there%20are%20two%20wildcard%20exports%20statements%20that%20implicitly%20re%2Dexport%20the%20same%20name%2C%20neither%20one%20is%20re%2Dexported.)

I think this case is hard to handle on Vite SSR model, so the behavior here is that first `export *` wins.